src/main/java/org/zalando/failsafeactuator/domain/FailsafeBreaker.java

```diff
@@ -0,0 +1,40 @@
+package org.zalando.failsafeactuator.domain;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import lombok.Data;
+import net.jodah.failsafe.CircuitBreaker;
+
+@Data
+public class FailsafeBreaker extends CircuitBreaker {
+
+    private final Meter failureMeter;
+    private final Meter successMeter;
+    private final Meter overallMeter;
+    private String identifier;
+
+    public FailsafeBreaker(final MetricRegistry metricRegistry, final String identifier) {
```

Comment:
I'm thinking if we should provide an abstraction layer around the metrics registry that would allow the users to plugin a different metrics collector, eg. prometheus.

For doing so we would need to introduce apart from the registry interface a Meter interface as well which would encapsulate the meter than.

As the default I would propose to provide a DropWizard Metrics implementation.

WDYT?

Reply:
Spring supports [PublicMetrics](https://github.com/spring-projects/spring-boot/blob/v1.5.7.RELEASE/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/PublicMetrics.java) for this.